godot/ModularWeapons/PlayerShip/PlayerShip.gd

```diff
@@ -1,44 +1,50 @@
-extends KinematicBody2D
+extends CharacterBody2D
 
-export var drag_linear_coeff := 0.05
-export var drag_angular_coeff := 0.1
+@export var drag_linear_coeff := 0.05
+@export var drag_angular_coeff : float = 0.1
 
 var acceleration_max := 1200.0
 var linear_speed_max := 540.0
 var angular_speed_max := 200
 var angular_acceleration_max := 1800
 
 var linear_velocity := Vector2.ZERO
-var angular_velocity := 0.0
+var angular_velocity : float = 0.0
 
-onready var weapons := $ModularWeaponsSystem
+@onready var weapons := $ModularWeaponsSystem
 
 
 func _ready() -> void:
-	var _error := weapons.connect("damaged", self, "_on_Weapons_damaged")
+	var _error := weapons.connect("damaged",Callable(self,"_on_Weapons_damaged"))
```

Comment:
Connect directly to the function: `var _error := weapons.connect("damaged", _on_Weapons_damaged)`